app/src/pages/Home.tsx

```diff
@@ -4,30 +4,38 @@ import axios from 'axios';
 import RequestsTable from '../components/custom/tables/RequestsTable';
 import { headerFont } from '../constants/fonts';
 import { useAuthService } from '../keycloak';
+import { useNavigate } from 'react-router-dom';
 
 /**
  * @description The Home page, showing a list of reimbursement requests. 
  * @returns A React element
  */
 const Home = () => {
   const [data, setData] = useState([]);
-  const { BACKEND_URL } = Constants;
+  const { BACKEND_URL, FRONTEND_URL } = Constants;
   const { state: authState } = useAuthService();
+  const navigate = useNavigate();
 
   // Fires on page load.
   useEffect(() => {
-    (async () => {
-      try {
-        const { data } = await axios.get(`${BACKEND_URL}/api/requests?minimal=true`, {
-          headers: {
-            Authorization : `Bearer ${authState.accessToken}`
-          }
-        })
-        setData(data);
-      } catch (e) {
-        console.warn('Server could not be reached.');
-      }
-    })();
+    const targetPage = sessionStorage.getItem('target-page');
+    if (targetPage && targetPage !== FRONTEND_URL){
+      sessionStorage.clear()
+      navigate(targetPage);
+    } else {
+      (async () => {
+        try {
+          const { data } = await axios.get(`${BACKEND_URL}/api/requests?minimal=true`, {
```

Comment:
Could this IIFE be refactored out into an async useCallback function?

Reply:
Updated to include a useCallback. It seems to be working, but not sure if I'm doing this like you expected.
Also changed the state variable names to avoid confusion with `data` returned from the axios call.